<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ThreeDViewerClass</class>
 <widget class="QMainWindow" name="ThreeDViewerClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>867</width>
    <height>581</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ThreeDViewer</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="QScrollArea" name="scrollArea">
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollAreaWidgetContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>507</width>
         <height>491</height>
        </rect>
       </property>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>867</width>
     <height>24</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionSave_as"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuImage">
    <property name="title">
     <string>Image</string>
    </property>
    <addaction name="actionClear"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuImage"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <widget class="QDockWidget" name="dockWidget">
   <property name="windowTitle">
    <string>Tools</string>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <widget class="QScrollArea" name="scrollArea_2">
       <property name="widgetResizable">
        <bool>true</bool>
       </property>
       <widget class="QWidget" name="scrollAreaWidgetContents_2">
        <property name="geometry">
         <rect>
          <x>0</x>
          <y>-168</y>
          <width>307</width>
          <height>865</height>
         </rect>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="QGroupBox" name="DISPLAY">
           <property name="title">
            <string>Display</string>
           </property>
           <layout class="QVBoxLayout" name="verticalLayout_2">
            <item>
             <widget class="QPushButton" name="global_color">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_2">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QComboBox" name="coloring_type">
              <item>
               <property name="text">
                <string>WireFrame</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Vertex</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Sides</string>
               </property>
              </item>
             </widget>
            </item>
            <item>
             <widget class="QComboBox" name="alg_type">
              <item>
               <property name="text">
                <string>DDA</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Bresenham</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="CAMERA">
           <property name="title">
            <string>Camera</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_2">
            <item row="5" column="1">
             <widget class="QPushButton" name="camera_front">
              <property name="text">
               <string>FRONT</string>
              </property>
             </widget>
            </item>
            <item row="5" column="2">
             <widget class="QPushButton" name="camera_right">
              <property name="text">
               <string>RIGHT</string>
              </property>
             </widget>
            </item>
            <item row="3" column="2">
             <widget class="QDoubleSpinBox" name="camera_z">
              <property name="minimum">
               <double>-1000.000000000000000</double>
              </property>
              <property name="maximum">
               <double>1000.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>y:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QDoubleSpinBox" name="camera_x">
              <property name="minimum">
               <double>-1000.000000000000000</double>
              </property>
              <property name="maximum">
               <double>1000.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="2" column="2">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>z:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QDoubleSpinBox" name="camera_y">
              <property name="minimum">
               <double>-1000.000000000000000</double>
              </property>
              <property name="maximum">
               <double>1000.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>x:</string>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QPushButton" name="camera_top">
              <property name="text">
               <string>TOP</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0" colspan="3">
             <widget class="QSlider" name="z_scale">
              <property name="mouseTracking">
               <bool>false</bool>
              </property>
              <property name="maximum">
               <number>100</number>
              </property>
              <property name="singleStep">
               <number>1</number>
              </property>
              <property name="value">
               <number>100</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="3">
             <widget class="QLabel" name="label_17">
              <property name="text">
               <string>z-coordinate scale</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="PROJECTIONS">
           <property name="title">
            <string>Projection type</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="1" column="0">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Center</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QDoubleSpinBox" name="center_of_projection">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="minimum">
               <double>1.000000000000000</double>
              </property>
              <property name="maximum">
               <double>1000.000000000000000</double>
              </property>
              <property name="value">
               <double>300.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="2">
             <widget class="QComboBox" name="projection_type">
              <item>
               <property name="text">
                <string>Ortographic</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>Perspective</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="LIGHT_SOURCE">
           <property name="title">
            <string>Point Light Source</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="2" column="2">
             <widget class="QDoubleSpinBox" name="light_z">
              <property name="minimum">
               <double>-1000.000000000000000</double>
              </property>
              <property name="maximum">
               <double>1000.000000000000000</double>
              </property>
              <property name="value">
               <double>400.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QDoubleSpinBox" name="light_y">
              <property name="minimum">
               <double>-1000.000000000000000</double>
              </property>
              <property name="maximum">
               <double>1000.000000000000000</double>
              </property>
              <property name="value">
               <double>400.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="3" column="1" colspan="2">
             <widget class="QSlider" name="light_intensity">
              <property name="maximum">
               <number>100</number>
              </property>
              <property name="value">
               <number>50</number>
              </property>
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="0" column="0" colspan="3">
             <widget class="QPushButton" name="light_color">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLabel" name="label_7">
              <property name="text">
               <string>z:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>x:</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>y:</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>Intensity</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QDoubleSpinBox" name="light_x">
              <property name="minimum">
               <double>-1000.000000000000000</double>
              </property>
              <property name="maximum">
               <double>1000.000000000000000</double>
              </property>
              <property name="value">
               <double>400.000000000000000</double>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>Phong light model</string>
           </property>
           <layout class="QGridLayout" name="gridLayout">
            <item row="1" column="2">
             <widget class="QSpinBox" name="diffuse_red">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>127</number>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QSpinBox" name="ambient_blue">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>127</number>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <widget class="QLabel" name="label_14">
              <property name="text">
               <string>Mirror</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>Green</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_10">
              <property name="text">
               <string>Blue</string>
              </property>
             </widget>
            </item>
            <item row="6" column="2" colspan="2">
             <widget class="QPushButton" name="ambient_color">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_12">
              <property name="text">
               <string>Red</string>
              </property>
             </widget>
            </item>
            <item row="4" column="0" colspan="2">
             <widget class="QLabel" name="label_15">
              <property name="layoutDirection">
               <enum>Qt::LeftToRight</enum>
              </property>
              <property name="text">
               <string>Mirror sharpness</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
            <item row="2" column="3">
             <widget class="QSpinBox" name="mirror_green">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>127</number>
              </property>
             </widget>
            </item>
            <item row="4" column="2" colspan="2">
             <widget class="QDoubleSpinBox" name="mirror_sharpness">
              <property name="minimum">
               <double>1.000000000000000</double>
              </property>
              <property name="maximum">
               <double>10.000000000000000</double>
              </property>
              <property name="singleStep">
               <double>0.100000000000000</double>
              </property>
              <property name="value">
               <double>1.000000000000000</double>
              </property>
             </widget>
            </item>
            <item row="3" column="2">
             <widget class="QSpinBox" name="diffuse_blue">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>127</number>
              </property>
             </widget>
            </item>
            <item row="3" column="3">
             <widget class="QSpinBox" name="mirror_blue">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>127</number>
              </property>
             </widget>
            </item>
            <item row="1" column="3">
             <widget class="QSpinBox" name="mirror_red">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>127</number>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="ambient_green">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>127</number>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="label_13">
              <property name="text">
               <string>Diffuse</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="ambient_red">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>127</number>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string>Ambient</string>
              </property>
             </widget>
            </item>
            <item row="2" column="2">
             <widget class="QSpinBox" name="diffuse_green">
              <property name="maximum">
               <number>255</number>
              </property>
              <property name="value">
               <number>127</number>
              </property>
             </widget>
            </item>
            <item row="6" column="0" colspan="2">
             <widget class="QLabel" name="label_16">
              <property name="text">
               <string>Ambient color</string>
              </property>
              <property name="alignment">
               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>40</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave_as">
   <property name="text">
    <string>Save as</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>Clear</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
   <property name="shortcut">
    <string>Alt+F4</string>
   </property>
  </action>
  <action name="actionResize">
   <property name="text">
    <string>Resize</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
